spec/exporters/patients_with_history_exporter_spec.rb

```diff
@@ -0,0 +1,266 @@
+require "rails_helper"
+
+RSpec.describe PatientsWithHistoryExporter do
+  include QuarterHelper
+
+  let!(:now) { Time.current }
+
+  let!(:patient) { create(:patient, status: "dead") }
+  let!(:facility) { patient.registration_facility }
+  let!(:user) { patient.registration_user }
+
+  let!(:bp_1) { create(:blood_pressure, :with_encounter, :critical, recorded_at: 2.months.ago, facility: facility, patient: patient, user: user) }
+  let!(:blood_sugar) { create(:blood_sugar, :fasting, :with_encounter, recorded_at: 2.months.ago, facility: facility, patient: patient, user: user) }
+  let!(:bp_1_follow_up) { create(:appointment, :overdue, device_created_at: 2.months.ago, scheduled_date: 40.days.ago, creation_facility: facility, patient: patient, user: user) }
+
+  let!(:bp_2) { create(:blood_pressure, :with_encounter, recorded_at: 3.months.ago, facility: facility, patient: patient, user: user) }
+  let!(:old_blood_sugar) { create(:blood_sugar, :with_encounter, recorded_at: 3.months.ago, facility: facility, patient: patient, user: user) }
+  let!(:bp_2_follow_up) { create(:appointment, device_created_at: 3.months.ago, scheduled_date: 2.months.ago, creation_facility: facility, patient: patient, user: user) }
+
+  let!(:bp_3) { create(:blood_pressure, :with_encounter, recorded_at: 4.months.ago, facility: facility, patient: patient, user: user) }
+  let!(:bp_3_follow_up) { create(:appointment, device_created_at: 4.month.ago, scheduled_date: 3.months.ago, creation_facility: facility, patient: patient, user: user) }
+
+  let!(:prescription_drugs) do
+    [
+      *create_list(:prescription_drug,
+        4,
+        :protocol,
+        device_created_at: 3.months.ago,
+        facility: facility,
+        patient: patient).sort_by(&:name),
+      *create_list(:prescription_drug,
+        3,
+        device_created_at: 3.months.ago,
+        facility: facility,
+        patient: patient).sort_by(&:name)
+    ]
+  end
```

Comment:
We should introduce a lot more nuance to the test data we set up here. Here are some thoughts:

* Create drugs on specific appointment days
* Delete some drugs on appointment days

Reply:
The drugs test are already covered in `prescription_drug_spec`. A couple of cases I'd like to add here though are:
- Have a `bp_4`
- Have medications on `bp_4` and make sure they show up correctly.

Reply:
Ah great points, and agree with the holes you've suggested filling 👍